src/aiida/cmdline/commands/cmd_daemon.py

```diff
@@ -217,24 +218,18 @@ def stop(ctx, no_wait, all_profiles, timeout):
 def restart(ctx, reset, no_wait, timeout):
     """Restart the daemon.
 
-    By default will only reset the workers of the running daemon. After the restart the same amount of workers will be
-    running. If the `--reset` flag is passed, however, the full daemon will be stopped and restarted with the default
+    By default the full daemon will be stopped and restarted with the default
     number of workers that is started when calling `verdi daemon start` manually.
 
     Returns exit code 0 if the result is OK, non-zero if there was an error.
     """
     if reset:
-        # These two lines can be simplified to `ctx.invoke(start)` once issue #950 in `click` is resolved.
-        # Due to that bug, the `callback` of the `number` argument the `start` command is not being called, which is
-        # responsible for settting the default value, which causes `None` to be passed and that triggers an exception.
-        # As a temporary workaround, we fetch the default here manually and pass that in explicitly.
-        number = ctx.obj.config.get_option('daemon.default_workers', ctx.obj.profile.name)
-        ctx.invoke(stop)
-        ctx.invoke(start, number=number)
-        return
-
-    echo.echo('Restarting the daemon... ', nl=False)
-    execute_client_command('restart_daemon', wait=not no_wait, timeout=timeout)
+        warn_deprecation(
+            '`--reset` flag is deprecated. Now, `verdi daemon restart` by default restarts the full daemon.', nl=False
+        )
+
+    ctx.invoke(stop, no_wait=no_wait)
```

Comment:
What would be the interest for a user to not wait here? Maybe we should also deprecate the `--no-wait` and `--timeout` options?

Reply:
Yes, it's better to deprecate, those as well.

Reply:
I mean now that you explained, I understand there is no need for these two flags anymore. I cannot imagine a use case for them.

Reply:
Something like this?

```suggestion
    if no_wait:
        warn_deprecation(
            '`--no_wait` flag is deprecated along with `--reset`. This flag is useless because workers alone cannot be reset, anymore.', version=3
        )
    if timeout != None:
        warn_deprecation(
            '`timeout` ignored. It\'s deprecated along with `--no_wait` and `--reset`.', version=3
        )
        
    ctx.invoke(stop,)
```

Reply:
I would keep the deprecation messages simple. No need to cross reference the other flags. Something like
```python
if no_wait:
    warn_deprecation('The `--no-wait` flag is deprecated and no longer has any effect.')

if timeout is not None:
    warn_deprecation('The `--timeout` option is deprecated and no longer has any effect.')
```